File: frontend/src/api/analysis.ts
import apiClient from './client'

export interface AnalysisJob {
  id: string
  novel_title: string
  source_file: string
  status: string  // pending | scanning | extracting | aggregating | done | failed
  progress: number
  chapter_count: number
  result_summary: Record<string, any>
  error_message: string
  created_at: string | null
  finished_at: string | null
}

export interface AnalysisChapter {
  id: string
  job_id: string
  chapter_number: number
  chapter_title: string
  characters: any[]
  events: any[]
  relationships: any[]
  foreshadows: any[]
  summary: string
  word_count: number
}

export const analysisApi = {
  listJobs: () => apiClient.get<AnalysisJob[]>('/analysis/jobs').then(r => r.data),

  getJob: (jobId: string) => apiClient.get<AnalysisJob>(`/analysis/jobs/${jobId}`).then(r => r.data),

  getJobChapters: (jobId: string) =>
    apiClient.get<AnalysisChapter[]>(`/analysis/jobs/${jobId}/chapters`).then(r => r.data),

  upload: (file: File) => {
    const form = new FormData()
    form.append('file', file)
    return apiClient.post<AnalysisJob>('/analysis/upload', form, {
      headers: { 'Content-Type': 'multipart/form-data' },
      timeout: 120000,
    }).then(r => r.data)
  },

  deleteJob: (jobId: string) => apiClient.delete(`/analysis/jobs/${jobId}`),
}

File: backend/app/schemas/analysis.py
"""拆书分析 Pydantic DTO"""
from __future__ import annotations

from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class AnalysisJobCreate(BaseModel):
    novel_title: str
    source_file: str


class AnalysisJobDTO(BaseModel):
    id: str
    novel_title: str
    source_file: str
    status: str
    progress: float
    chapter_count: int
    result_summary: dict
    error_message: str
    created_at: datetime
    finished_at: Optional[datetime]

    class Config:
        from_attributes = True


class AnalysisChapterDTO(BaseModel):
    id: str
    job_id: str
    chapter_number: int
    chapter_title: str
    characters: list[dict]
    events: list[dict]
    relationships: list[dict]
    foreshadows: list[dict]
    summary: str
    word_count: int

    class Config:
        from_attributes = True
